commit 3da59d8f99863d1c4d90fa9fcbc33721ec7cebc1
Author: Mario Haustein <mario.haustein@hrz.tu-chemnitz.de>
Date:   Wed Oct 5 12:42:11 2022 +0200

    Fix Celestrak file list fetching logic
    
    Celestrak changed the link references on their website. We need to
    account for it.

diff --git a/space/tle/celestrak.py b/space/tle/celestrak.py
index 3c2aa77..e557b83 100644
--- a/space/tle/celestrak.py
+++ b/space/tle/celestrak.py
@@ -2,6 +2,7 @@ import logging
 import asyncio
 import aiohttp
 import async_timeout
+import re
 import requests
 from bs4 import BeautifulSoup
 
@@ -84,8 +85,10 @@ def fetch_list():
     files = []
     bs = BeautifulSoup(page.text, features="lxml")
     for link in bs.body.find_all("a"):
-        if "href" in link.attrs and link["href"].endswith(".txt"):
-            files.append(link.get("href"))
+        if "href" in link.attrs:
+            linkmatch = re.fullmatch(r"gp\.php\?GROUP=([A-Za-z0-9\-]+)&FORMAT=tle", link["href"])
+            if linkmatch is not None:
+                files.append(linkmatch.group(1) + ".txt")
 
     log.info("%d celestrak files found", len(files))
 

commit 77a203a23df902b4dd82cf279bf024d32f41f398
Author: Mario Haustein <mario.haustein@hrz.tu-chemnitz.de>
Date:   Wed Oct 5 12:41:37 2022 +0200

    Update Celestrak domain

diff --git a/space/tle/celestrak.py b/space/tle/celestrak.py
index e995885..3c2aa77 100644
--- a/space/tle/celestrak.py
+++ b/space/tle/celestrak.py
@@ -12,7 +12,7 @@ from .db import TleDb
 log = logging.getLogger(__name__)
 
 TMP_FOLDER = TMP_FOLDER / "celestrak"
-CELESTRAK_URL = "http://celestrak.com/NORAD/elements/"
+CELESTRAK_URL = "http://celestrak.org/NORAD/elements/"
 PAGE_LIST_CONFIG = ("celestrak", "page-list")
 DEFAULT_FILES = [
     "stations.txt",
